swingset/src/main/java/com/nqadmin/swingset/models/SSDbArrayModel.java

```diff
@@ -87,7 +87,11 @@ public SSDbArrayModel(final JDBCType _jdbcType) {
 	/** {@inheritDoc } */
 	@Override
 	public Object[] readData(final RowSet _rowSet, final String _columnName) throws SQLException {
-		return toObjList(getJDBCType(), _rowSet.getArray(_columnName)).toArray();
+		List<Object> data = toObjList(getJDBCType(), _rowSet.getArray(_columnName));
+		if(data == null) {
+			return null;
```

Comment:
toObjList starts with a check an may return null. So definitely a bug that needed to be fixed here. I didn't look at the callers of readData to see if returning an empty array would be better.